src/azure-cli-core/azure/cli/core/application.py

```diff
@@ -44,6 +44,23 @@ def get_command_table(self, argv=None):  # pylint: disable=no-self-use
         matches = Configuration.find_matches(argv, command_tree)
         return dict(matches)
 
+    # raise customized exceptions on removed commands if the command module like
+    @staticmethod
+    def err_on_removed_command(argv):
+        import azure.cli.core.commands as commands
```

Comment:
This routine can be further optimized if needed, but so far it doesn't cause noticeable perf overheads, as mostly are a few simple string  ops.